vault/core.go

```diff
@@ -2946,3 +2946,25 @@ type LicenseState struct {
 	ExpiryTime time.Time
 	Terminated bool
 }
+
+type PeerNode struct {
+	Hostname       string    `json:"hostname"`
+	APIAddress     string    `json:"api_address"`
+	ClusterAddress string    `json:"cluster_address"`
+	LastEcho       time.Time `json:"last_echo"`
+}
```

Comment:
This is the third incarnation of a node-like struct in this PR. Is it worth unifying all of these into 1 node struct that can be used in all 3 places?

Reply:
I tend to prefer having more specific types than having types with fields that may or may not be populated.  So PeerNode just describes what we've got from our peers' echo requests: it exists so that the Core method can return exactly that and no more.

http.Node defines the api endpoint's response; aside from my concern about unpopulated fields above, it's usually a good idea to *not* use the same type for internal purposes that you return to the user.

Finally, the type in the command pkg is the type that would appear in the api package - you'll see that we have this duplication everywhere.  I'm just punting on creating the api method for this endpoint, so it lives here instead fo rnow.

Reply:
Fair enough